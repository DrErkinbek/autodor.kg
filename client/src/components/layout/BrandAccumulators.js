import React, { useState } from 'react';
import exideBr from '../images/brand-exide.png';
import bostBr from '../images/brand-bost.png';
import tuborBr from '../images/brand-tubor.png';
import istaBr from '../images/brand-ista.png';
import gladiatorBr from '../images/brand-gladiator.png';
import kainarBr from '../images/brand-kainar.png';

const BrandAccumulators = () => {
    const [brands] = useState({
        exide: 'exide',
        bost: 'bost',
        tubor: 'tubor',
        ista: 'ista',
        gladiator: 'gladiator',
        kainar: 'kainar'
    });
    const {exide, bost, tubor, ista, gladiator, kainar } = brands;
    
    return(
        <>
        <div className="row">
            <div className="col s6 m3 l3 xl3">
                <div className="card">
                    <div className="card-image">
                        <a href={`/brands/${exide}`}>
                            <img src={exideBr} alt="exide" />
                        </a>
                    </div>
                    <div className="card-content center">
                        <a href={`/brands/${exide}`}>
                            <p className="blue-text product-title">EXIDE Европа</p>
                        </a>
                    </div>
                </div>
            </div>
            <div className="col s6 m3 l3 xl3">
                <div className="card">
                    <div className="card-image">
                        <a href={`/brands/${bost}`}>
                            <img src={bostBr} alt="exide" />
                        </a>
                    </div>
                    <div className="card-content center">
                        <a href={`/brands/${bost}`}>
                            <p className="blue-text product-title">BOST Корея</p>
                        </a>
                    </div>
                </div>
            </div>
            <div className="col s6 m3 l3 xl3">
                <div className="card">
                    <div className="card-image">
                        <a href={`/brands/${tubor}`}>
                            <img src={tuborBr} alt="exide" />
                        </a>
                    </div>
                    <div className="card-content center">
                        <a href={`/brands/${tubor}`}>
                            <p className="blue-text product-title">TUBOR Россия</p>
                        </a>
                    </div>
                </div>
            </div>
            <div className="col s6 m3 l3 xl3">
                <div className="card">
                    <div className="card-image">
                        <a href={`/brands/${ista}`}>
                            <img src={istaBr} alt="exide" />
                        </a>
                    </div>
                    <div className="card-content center">
                        <a href={`/brands/${ista}`}>
                            <p className="blue-text product-title">ISTA Украина</p>
                        </a>
                    </div>
                </div>
            </div>
        </div>
        <div className="row">
            <div className="col s6 m3 l3 xl3">
                <div className="card">
                    <div className="card-image">
                        <a href={`/brands/${gladiator}`}>
                            <img src={gladiatorBr} alt="exide" />
                        </a>
                    </div>
                    <div className="card-content center">
                        <a href={`/brands/${gladiator}`}>
                            <p className="blue-text product-title">GLADIATOR Россия</p>
                        </a>
                    </div>
                </div>
            </div>
            <div className="col s6 m3 l3 xl3">
                <div className="card">
                    <div className="card-image">
                        <a href={`/brands/${kainar}`}>
                            <img src={kainarBr} alt="exide" />
                        </a>
                    </div>
                    <div className="card-content center">
                        <a href={`/brands/${kainar}`}>
                            <p className="blue-text product-title">KAINAR Казакстан</p>
                        </a>
                    </div>
                </div>
            </div>
        </div>
        </>
    )
}

export default BrandAccumulators;