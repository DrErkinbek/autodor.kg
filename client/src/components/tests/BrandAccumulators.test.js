import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import BrandAccumulators from '../layout/BrandAccumulators';

it('BrandAccumulators Component renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrandAccumulators />, div);
});

it('BrandAccumulators Component renders correctly and equal to snapshot', () => {
    const tree = renderer.create(<BrandAccumulators />).toJSON();
  
    expect(tree).toMatchSnapshot();
});