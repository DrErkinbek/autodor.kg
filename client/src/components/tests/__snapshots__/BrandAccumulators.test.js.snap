// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandAccumulators Component renders correctly and equal to snapshot 1`] = `
Array [
  <div
    className="row"
  >
    <div
      className="col s6 m3 l3 xl3"
    >
      <div
        className="card"
      >
        <div
          className="card-image"
        >
          <img
            alt="exide"
            src="brand-exide.png"
          />
        </div>
        <div
          className="card-content center"
        >
          <a
            href="/brands/exide"
          >
            <p
              className="blue-text product-title"
            >
              EXIDE Европа
            </p>
          </a>
        </div>
      </div>
    </div>
    <div
      className="col s6 m3 l3 xl3"
    >
      <div
        className="card"
      >
        <div
          className="card-image"
        >
          <img
            alt="exide"
            src="brand-bost.png"
          />
        </div>
        <div
          className="card-content center"
        >
          <a
            href="/brands/bost"
          >
            <p
              className="blue-text product-title"
            >
              BOST Корея
            </p>
          </a>
        </div>
      </div>
    </div>
    <div
      className="col s6 m3 l3 xl3"
    >
      <div
        className="card"
      >
        <div
          className="card-image"
        >
          <img
            alt="exide"
            src="brand-tubor.png"
          />
        </div>
        <div
          className="card-content center"
        >
          <a
            href="/brands/tubor"
          >
            <p
              className="blue-text product-title"
            >
              TUBOR Россия
            </p>
          </a>
        </div>
      </div>
    </div>
    <div
      className="col s6 m3 l3 xl3"
    >
      <div
        className="card"
      >
        <div
          className="card-image"
        >
          <img
            alt="exide"
            src="brand-ista.png"
          />
        </div>
        <div
          className="card-content center"
        >
          <a
            href="/brands/ista"
          >
            <p
              className="blue-text product-title"
            >
              ISTA Украина
            </p>
          </a>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col s6 m3 l3 xl3"
    >
      <div
        className="card"
      >
        <div
          className="card-image"
        >
          <img
            alt="exide"
            src="brand-gladiator.png"
          />
        </div>
        <div
          className="card-content center"
        >
          <a
            href="/brands/gladiator"
          >
            <p
              className="blue-text product-title"
            >
              GLADIATOR Россия
            </p>
          </a>
        </div>
      </div>
    </div>
    <div
      className="col s6 m3 l3 xl3"
    >
      <div
        className="card"
      >
        <div
          className="card-image"
        >
          <img
            alt="exide"
            src="brand-kainar.png"
          />
        </div>
        <div
          className="card-content center"
        >
          <a
            href="/brands/kainar"
          >
            <p
              className="blue-text product-title"
            >
              KAINAR Казакстан
            </p>
          </a>
        </div>
      </div>
    </div>
  </div>,
]
`;
